JobsPortal/src/main/java/com/ps/js/controller/JobsController.java

```diff
@@ -0,0 +1,363 @@
+		Set<RolesAndResponsebility> rolesAndResponsebilities = new HashSet<RolesAndResponsebility>();
+		JobDetails jobDetails = new JobDetails();
+		JobDetailsResponsePayload jobDetailsResponsePayload = null;
+		// Convert job details to job details payload using jobDetailsMapper service
+		jobDetails = jobDetailsMapper.jobDetailsPayloadToJobDetailsMapper(jobDetailsPayload, jobDetails);
+
+		System.out.println("jobDetails :: " + jobDetails);
+
+		// Fetching the location and adding to collection
+		for (JobLocation location : jobDetails.getJobLocation()) {
+			locations.add(jobLocationServices.fetchJobLocationById(location.getLocationId()).get());
+		}
+
+		// Fetching the primary skills and added to collection
+		for (Skill skill : jobDetails.getPrimarySkill()) {
+			primarySkills.add(skillServices.fetchSkillById(skill.getSkillId()).get());
+		}
+
+		// Fetching the secondary skills and added to collection
+		for (Skill skill : jobDetails.getSecondrySkill()) {
+			secondrySkills.add(skillServices.fetchSkillById(skill.getSkillId()).get());
+		}
+
+		// Fetching the roles and responsebility and added to collection
+		for (RolesAndResponsebility responsebility : jobDetails.getRolesAndResponsebilities()) {
+			rolesAndResponsebilities
+					.add(rolesAndResponsebilityService.fetchRolesAndResponsebilityById(responsebility.getId()).get());
+		}
+		if (locations.isEmpty())
+			throw new LocationsNotFoundException();
+		else if (primarySkills.isEmpty())
+			throw new SkillNotFoundException(ErrorMessages.PRIMARY_SKILL_NOT__FOUND_EXCEPTION.toString());
+		else if (secondrySkills.isEmpty())
+			throw new SkillNotFoundException(ErrorMessages.SECONDRY_SKILL_NOT__FOUND_EXCEPTION.toString());
+
+		// Invoking jobDetailsService to add a new jobDetails
+		jobDetailsResponsePayload = jobDetailsService.createJob(jobDetails);
+
+		if (jobDetailsResponsePayload != null) {
+			for (ResponsePayload payload : jobDetailsResponsePayload.getJobLocation()) {
+				payload.setUrl("jobs/find-location-by-id/" + payload.getId());
+			}
+			for (ResponsePayload payload : jobDetailsResponsePayload.getPrimarySkill()) {
+				payload.setUrl("jobs/find-skill-by-id/" + payload.getId());
+			}
+			for (ResponsePayload payload : jobDetailsResponsePayload.getSecondrySkill()) {
+				payload.setUrl("jobs/find-skill-by-id/" + payload.getId());
+			}
+			for (ResponsePayload payload : jobDetailsResponsePayload.getRolesAndResponsebilities()) {
+				payload.setUrl("jobs/find-roles-and-responsebility-by-id/" + payload.getId());
+			}
+		}
+
+		// System.out.println("jobDetailsResponsePayload "+jobDetailsResponsePayload);
+
+		return new ResponseEntity<JobDetailsResponsePayload>(jobDetailsResponsePayload, HttpStatus.OK);
+	}
+
+	/**
+	 * To find skill by using skill-id
+	 * 
+	 * @param skillId
+	 * @return {@link ResponseEntity<Skill>}
+	 */
+	@GetMapping("/find-skill-by-id/{skill-id}")
+	public ResponseEntity<Skill> findSKillById(@PathVariable("skill-id") int skillId) {
```

Comment:
Separate controller for Jobs (CRUD operations), Separate  Controller for Skills, the same way for Locations, roles and all!
One controller should have crud operations related to one entity/schema/element you call it any name but one class one job (SOLID principles)

Reply:
mapping should be only find-by-skill/{skill-id}